translator/writer/prim_writer.cpp

```diff
@@ -1157,8 +1157,18 @@ static void processMaterialBinding(AtNode* shader, AtNode* displacement, UsdPrim
 
     TfToken arnoldContext("arnold");
     if (shader) {
-        // write the surface shader under the material's scope
+        // Write the surface shader under the material's scope.
+        // Here we only want to consider the last name in the prim 
+        // hierarchy, so we're stripping the scope here
+        size_t npos = shaderName.rfind('/');
```

Comment:
I am wondering if, instead of using the latest name component, we should just strip the scope that is going to be added later on ? this would allow to have scopes within the current material scope.
I haven't tested but I am thinking something like the following:
```
if (TfStringStartsWith(shaderName, materialName)) {
  std::string nodeLastName = shaderName.substr(materialName.size());
  ....
}
```